contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -210,35 +205,23 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
     /**
      * @dev See {INativeTokenDestination-reportTotalBurnedTxFees}.
      */
-    function reportTotalBurnedTxFees(
-        TeleporterFeeInfo calldata feeInfo,
-        address[] calldata allowedRelayerAddresses
-    ) external {
-        uint256 adjustedFeeAmount;
-        if (feeInfo.amount > 0) {
-            adjustedFeeAmount = SafeERC20TransferFrom.safeTransferFrom(
-                IERC20(feeInfo.feeTokenAddress), feeInfo.amount
-            );
-        }
+    function reportBurnedTxFees(address[] calldata allowedRelayerAddresses) external {
```

Comment:
Comment for future consideration in a later PR:

I think we should consider having `allowedRelayerAddresses` be something that is set in Teleporter application constructors, with the managers of the application being allowed to update it. IMO, the use case for `allowedRelayers` is if an application wants to restrict who is able to relay it's messages to provide off-chain checks (on validity, compliance, etc). I don't think it makes sense for message senders (i.e. end users calling the applications) to be able to set it themselves, since they only care about anyone delivering their message, but not who.

Reply:
cc @minghinmatthewlam for you to think about also as we move forward with the Teleporter bridge contract implementations.

Reply:
Yeah I do like the idea of abstracting it away from the users.

Reply:
would this make sense in `TeleporterUpgradeable`? Or do we want to leave it up to the dApps to implement and not set as a default

Reply:
hmm what happens for a contract that gets called by other dApps, who have their individual allowed relayer addresses though?

Reply:
> would this make sense in TeleporterUpgradeable? Or do we want to leave it up to the dApps to implement and not set as a default

I think it likely makes sense as a separate utility. Along the lines of a `RestrictedRelayerSender`.

> hmm what happens for a contract that gets called by other dApps, who have their individual allowed relayer addresses though?

Sorry not sure I follow. The dApp calling `sendCrossChainMessage` is what gets to define the allowed relayers for its own messages since they may have their own restrictions.

Reply:
Lets say we have contract A that gets called by a bunch of other contracts B, C, D, E...If contract A constructor takes in empty allowed relayers list, so any relayers allowed, but the other contracts (B,C,D...) have their own `allowedRelayerAddresses` list they want to pass down to the call of contract A.

Reply:
Reading over again, I think i'm in agreement for dApps that want to restrict relayers, they can set it in their constructor. For dApps that don't care about restricted relayers, they can still take in `allowedRelayerAddresses` as a parameter in sends.

Reply:
Yeah I think it makes sense for the immediate caller of `sendCrossChainMessage` (i.e. the cross-chain dApp itself) to limit the allowed relayers if necessary. Exposing that option to callers of the dApp makes it possible for users of the cross-chain dApp to limit relayers for their own messages, but that's not something I would generally recommend for dApps, since users should not need to know of or about relayers at all.